File: litsea-python/examples/segment.py
"""Segment a sentence, with and without POS tags.

Usage:
    python examples/segment.py ../models/japanese.model "これはテストです。"
    python examples/segment.py ../models/japanese_pos.model "これはテストです。"
"""

from __future__ import annotations

import sys

from litsea import Segmenter


def main() -> int:
    """Run the example.

    Returns:
        0 on success, 2 when the arguments are wrong.
    """
    if len(sys.argv) != 3:
        print(__doc__, file=sys.stderr)
        return 2

    model_path, text = sys.argv[1], sys.argv[2]

    # The model file identifies its own kind, so nothing here says "this is
    # a POS model" - `has_pos` reports what was loaded.
    segmenter = Segmenter.open("japanese", model_path)
    print(f"model: {model_path} (has_pos={segmenter.has_pos})")

    print("tokens:", " ".join(segmenter.segment(text)))

    if segmenter.has_pos:
        print("tagged:")
        for token in segmenter.segment_with_pos(text):
            print(f"  {token.surface}\t{token.pos.name}\t[{token.start}:{token.end}]")

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: litsea-php/examples/segment.php
<?php

declare(strict_types=1);

/**
 * Segment a sentence, with and without POS tags.
 *
 * Usage:
 *   php -d extension=../target/release/liblitsea_php.so \
 *       examples/segment.php ../models/japanese.model "これはテストです。"
 */

if ($argc !== 3) {
    fwrite(STDERR, "usage: php examples/segment.php <model> <text>\n");
    exit(2);
}

[, $modelPath, $text] = $argv;

// The model file identifies its own kind, so nothing here declares whether
// this is a POS model - hasPos() reports what was loaded.
$segmenter = Litsea\Segmenter::open('japanese', $modelPath);
printf("model: %s (hasPos=%s)\n", $modelPath, $segmenter->hasPos() ? 'true' : 'false');
printf("tokens: %s\n", implode(' ', $segmenter->segment($text)));

if ($segmenter->hasPos()) {
    echo "tagged:\n";
    foreach ($segmenter->segmentWithPos($text) as $token) {
        printf("  %s\t%s\t[%d:%d]\n", $token->surface, $token->pos, $token->start, $token->end);
    }
}
